python/poetry/Portfile

```diff
@@ -0,0 +1,62 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem              1.0
+PortGroup               python 1.0
+
+name                    poetry
+version                 1.0.3
+revision                0
+categories-append       devel
+platforms               darwin
+license                 MIT
+supported_archs         noarch
+
+maintainers             {gmail.com:davidgilman1 @dgilman} openmaintainer
+
+description             Python dependency management and packaging made easy.
+
+long_description        Poetry: Dependency Management for Python \
+                        \
+                        Poetry helps you declare, manage and install \
+                        dependencies of Python projects, ensuring you have \
+                        the right stack everywhere.
+
+homepage                https://python-poetry.org/
+
+checksums               rmd160  d6094162b8f01e15d86fb40b7aaab7e15bce76f1 \
+                        sha256  60a079b1c1d8a4f1538d28ee8d1bf79a5d0eae96f497fdae9512eb3c1f8da13b \
+                        size    164894
+
+python.default_version  38
+
+# These are in setup.py order
+depends_lib-append \
+    port:py${python.version}-cachecontrol \
+    port:py${python.version}-lockfile \
+    port:py${python.version}-cachy \
+    port:py${python.version}-cleo \
+    port:py${python.version}-clikit \
+    port:py${python.version}-html5lib \
+    port:py${python.version}-jsonschema \
+    port:py${python.version}-pexpect \
+    port:py${python.version}-pkginfo \
+    port:py${python.version}-parsing \
+    port:py${python.version}-pyrsistent \
+    port:py${python.version}-requests-toolbelt \
+    port:py${python.version}-requests \
+    port:py${python.version}-shellingham \
+    port:py${python.version}-tomlkit \
+    port:py${python.version}-keyring
+
+post-extract {
+    reinplace "s|keyring>=20.0.1,<21.0.0|keyring>=20.0.1|" \
+                                                    ${worksrcpath}/setup.py
+    reinplace "s|requests-toolbelt>=0.8.0,<0.9.0|requests-toolbelt>=0.8.0|" \
+                                                    ${worksrcpath}/setup.py
+    reinplace "s|pyrsistent>=0.14.2,<0.15.0|pyrsistent>=0.14.2|" \
+                                                    ${worksrcpath}/setup.py
+}
+
+test.run                yes
+test.cmd                poetry
+test.target             about
```

Comment:
I am not really sure about this "test.cmd". Currently, it will only work if you've the package already installed - ideally it should not be needed; you likely need to the the PATH to make sure you can run this from the "build" directory or wherever the script is generated.

But, more importantly, I would say one should run upstream's test-suite here (admittedly that's not present in the PyPI archive, but getting the source from GitHub would resolve that). Especially, given the concern with the versioning-caps of dependencies I think it would be more helpful to run the full test-suite to make sure that everything works as intended, instead of only doing ```poetry about``` (which would only check whether the dependency versions are within their "allowed" ranges).

Reply:
I'm sort of taking a stab at this but I have a problem. I can use the pypi distfile and get a tar with no tests but a setup.py. Alternatively, I can use the github distfile and get a tar with tests but no setup.py. I am not sure if I can put together a Portfile that can be used for both building and testing.

Reply:
true... one other reason I am not a big fan of projects using ```poetry``` as of yet ;) The other option is to ask upstream to include the tests in the source file on PyPI.

Reply:
What about `python -m poetry` as a way to run the package? I disabled the poetry port before running that to confirm that I was running the poetry package that was in the work dir, but even if it wasn't disabled the Python import mechanism is going to use a package in the current working directory over one installed in site-packages.

Reply:
yes, that would work - you'd want to make sure you use the Python version that you've listed as "python.default_version" (currently 3.8). So likely ```test.cmd                ${python.bin} -m poetry``` is what you want here.

That's probably as much testing as one can do right now until upstream decides to add their tests to the PyPI files or we find another around the "setup.py versus tests dilemma".